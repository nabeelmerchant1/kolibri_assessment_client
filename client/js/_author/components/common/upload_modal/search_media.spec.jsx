import React            from 'react';
import { shallow }      from 'enzyme';
import SearchMedia      from './search_media';

describe('search media component', () => {
  let props;
  let result;

  beforeEach(() => {
    props = {
      media: {},
      loading: false,
      selectedMediaId: '',
      selectMedia: () => {},
    };
    result = shallow(<SearchMedia {...props} />);
  });

  it('matches the taken snapshot', () => {
    expect(result).toMatchSnapshot();
  });

  it('handles setting the state', () => {
    expect(result.instance().state.offset).toBe(0);
    const data = {
      selected: 5,
    };
    result.instance().handlePageClick(data);
    expect(result.instance().state.offset).toBe(40);
  });
});
