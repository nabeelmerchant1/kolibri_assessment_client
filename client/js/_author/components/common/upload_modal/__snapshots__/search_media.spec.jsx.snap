// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search media component matches the taken snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="au-c-wysiwyg-modal__media"
>
    <div
        className="au-c-modal-media__search"
    >
        <div
            className="au-c-input au-c-input--search"
        >
            <label
                htmlFor="name2"
            />
            <div
                className="au-c-input__contain"
            >
                <i
                    className="material-icons"
                >
                    search
                </i>
                <input
                    className="au-c-text-input au-c-text-input--small"
                    id="name2"
                    onChange={[Function]}
                    placeholder="Search..."
                    type="text"
                    value=""
                />
            </div>
        </div>
    </div>
    <mediaTable
        activeItem={undefined}
        media={Array []}
        selectMedia={[Function]}
        selectedMediaId=""
    />
    <PaginationBoxView
        activeClassName="is-active"
        breakClassName="au-c-modal-media__pagination-break"
        breakLabel={
            <span>
                ...
            </span>
        }
        containerClassName="au-c-modal-media__pagination"
        disableInitialCallback={false}
        disabledClassName="disabled"
        marginPagesDisplayed={1}
        nextClassName="au-c-modal-media__pagination-next "
        nextLabel={
            <i
                className="material-icons"
            >
                keyboard_arrow_right
            </i>
        }
        onPageChange={[Function]}
        pageClassName="au-c-modal-media__pagination-pages"
        pageCount={0}
        pageLinkClassName="au-c-modal-media__pagination-page"
        pageRangeDisplayed={5}
        previousClassName="au-c-modal-media__pagination-previous "
        previousLabel={
            <i
                className="material-icons"
            >
                keyboard_arrow_left
            </i>
        }
    />
</div>,
  "nodes": Array [
    <div
      className="au-c-wysiwyg-modal__media"
>
      <div
            className="au-c-modal-media__search"
      >
            <div
                  className="au-c-input au-c-input--search"
            >
                  <label
                        htmlFor="name2"
                  />
                  <div
                        className="au-c-input__contain"
                  >
                        <i
                              className="material-icons"
                        >
                              search
                        </i>
                        <input
                              className="au-c-text-input au-c-text-input--small"
                              id="name2"
                              onChange={[Function]}
                              placeholder="Search..."
                              type="text"
                              value=""
                        />
                  </div>
            </div>
      </div>
      <mediaTable
            activeItem={undefined}
            media={Array []}
            selectMedia={[Function]}
            selectedMediaId=""
      />
      <PaginationBoxView
            activeClassName="is-active"
            breakClassName="au-c-modal-media__pagination-break"
            breakLabel={
                  <span>
                        ...
                  </span>
            }
            containerClassName="au-c-modal-media__pagination"
            disableInitialCallback={false}
            disabledClassName="disabled"
            marginPagesDisplayed={1}
            nextClassName="au-c-modal-media__pagination-next "
            nextLabel={
                  <i
                        className="material-icons"
                  >
                        keyboard_arrow_right
                  </i>
            }
            onPageChange={[Function]}
            pageClassName="au-c-modal-media__pagination-pages"
            pageCount={0}
            pageLinkClassName="au-c-modal-media__pagination-page"
            pageRangeDisplayed={5}
            previousClassName="au-c-modal-media__pagination-previous "
            previousLabel={
                  <i
                        className="material-icons"
                  >
                        keyboard_arrow_left
                  </i>
            }
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchMedia
        loading={false}
        media={Object {}}
        selectMedia={[Function]}
        selectedMediaId=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchMedia {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "loading": false,
          "media": Object {},
          "selectMedia": [Function],
          "selectedMediaId": "",
        },
        "refs": Object {},
        "state": Object {
          "offset": 0,
          "searchText": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="au-c-wysiwyg-modal__media"
>
          <div
                    className="au-c-modal-media__search"
          >
                    <div
                              className="au-c-input au-c-input--search"
                    >
                              <label
                                        htmlFor="name2"
                              />
                              <div
                                        className="au-c-input__contain"
                              >
                                        <i
                                                  className="material-icons"
                                        >
                                                  search
                                        </i>
                                        <input
                                                  className="au-c-text-input au-c-text-input--small"
                                                  id="name2"
                                                  onChange={[Function]}
                                                  placeholder="Search..."
                                                  type="text"
                                                  value=""
                                        />
                              </div>
                    </div>
          </div>
          <mediaTable
                    activeItem={undefined}
                    media={Array []}
                    selectMedia={[Function]}
                    selectedMediaId=""
          />
          <PaginationBoxView
                    activeClassName="is-active"
                    breakClassName="au-c-modal-media__pagination-break"
                    breakLabel={
                              <span>
                                        ...
                              </span>
                    }
                    containerClassName="au-c-modal-media__pagination"
                    disableInitialCallback={false}
                    disabledClassName="disabled"
                    marginPagesDisplayed={1}
                    nextClassName="au-c-modal-media__pagination-next "
                    nextLabel={
                              <i
                                        className="material-icons"
                              >
                                        keyboard_arrow_right
                              </i>
                    }
                    onPageChange={[Function]}
                    pageClassName="au-c-modal-media__pagination-pages"
                    pageCount={0}
                    pageLinkClassName="au-c-modal-media__pagination-page"
                    pageRangeDisplayed={5}
                    previousClassName="au-c-modal-media__pagination-previous "
                    previousLabel={
                              <i
                                        className="material-icons"
                              >
                                        keyboard_arrow_left
                              </i>
                    }
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="au-c-wysiwyg-modal__media"
>
          <div
                    className="au-c-modal-media__search"
          >
                    <div
                              className="au-c-input au-c-input--search"
                    >
                              <label
                                        htmlFor="name2"
                              />
                              <div
                                        className="au-c-input__contain"
                              >
                                        <i
                                                  className="material-icons"
                                        >
                                                  search
                                        </i>
                                        <input
                                                  className="au-c-text-input au-c-text-input--small"
                                                  id="name2"
                                                  onChange={[Function]}
                                                  placeholder="Search..."
                                                  type="text"
                                                  value=""
                                        />
                              </div>
                    </div>
          </div>
          <mediaTable
                    activeItem={undefined}
                    media={Array []}
                    selectMedia={[Function]}
                    selectedMediaId=""
          />
          <PaginationBoxView
                    activeClassName="is-active"
                    breakClassName="au-c-modal-media__pagination-break"
                    breakLabel={
                              <span>
                                        ...
                              </span>
                    }
                    containerClassName="au-c-modal-media__pagination"
                    disableInitialCallback={false}
                    disabledClassName="disabled"
                    marginPagesDisplayed={1}
                    nextClassName="au-c-modal-media__pagination-next "
                    nextLabel={
                              <i
                                        className="material-icons"
                              >
                                        keyboard_arrow_right
                              </i>
                    }
                    onPageChange={[Function]}
                    pageClassName="au-c-modal-media__pagination-pages"
                    pageCount={0}
                    pageLinkClassName="au-c-modal-media__pagination-page"
                    pageRangeDisplayed={5}
                    previousClassName="au-c-modal-media__pagination-previous "
                    previousLabel={
                              <i
                                        className="material-icons"
                              >
                                        keyboard_arrow_left
                              </i>
                    }
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchMedia
    loading={false}
    media={Object {}}
    selectMedia={[Function]}
    selectedMediaId=""
/>,
}
`;
